src/router.ts

```diff
@@ -17,27 +18,41 @@ export function createRouter<T extends RadixNodeData = RadixNodeData>(
     staticRoutesMap: {},
   };
 
-  const normalizeTrailingSlash = (p) =>
+  const normalizeTrailingSlash = (p: string) =>
     options.strictTrailingSlash ? p : p.replace(/\/$/, "") || "/";
 
   if (options.routes) {
     for (const path in options.routes) {
-      insert(ctx, normalizeTrailingSlash(path), options.routes[path]);
+      Array.isArray(options.routes[path]) &&
+      options.routes[path].length === 2 &&
+      HTTPMethods.includes(options.routes[path][0])
+        ? insert(
+            ctx,
+            normalizeTrailingSlash(path),
+            options.routes[path][1],
+            options.routes[path][0]
+          )
+        : insert(ctx, normalizeTrailingSlash(path), options.routes[path]);
```

Comment:
This is not essential for the feature (as also mentioned in acceptance to be minimal). I think in **next** steps/PR we can think of introducing either object format or array (I prefer object) with easier discussions that don't block essential part.